### PR TITLE
Fixing issue with firebase init genkit on Windows

### PR DESCRIPTION
**Pull request by ifielker**
Fixing issue with firebase init genkit on Windows.

### CODE REVIEW

File: src/init/features/genkit/index.ts [hunk 1]
```diff
@@ -61,7 +59,18 @@ async function getGenkitVersion(): Promise<GenkitInfo> {
...
+    try {
+      genkitVersion = await spawnWithOutput("npm", ["view", "genkit", "version"]);
+    } catch (err: unknown) {
+      throw new FirebaseError(
+        "Unable to determine which genkit version to install.\n" +
+          `npm Error: ${getErrMsg(err)}\n\n` +
+          "For a possible workaround run\n  npm view genkit version\n" +
+          "and then set an environment variable:\n" +
+          "  export GENKIT_DEV_VERSION=<output from previous command>\n" +
+          "and try the firebase init command again",
```
Comment (joehan) [inline src/init/features/genkit/index.ts:71]:
```suggestion
          "and run `firebase init` again",
```

File: src/extensions/runtimes/node.ts [hunk 2]
```diff
@@ -482,7 +483,7 @@ export async function writeSDK(
   // Build it
   logLabeledBullet("extensions", `running 'npm --prefix ${shortDirPath} run build'`);
   try {
-    execFileSync("npm", ["--prefix", dirPath, "run", "build"]);
+    await spawnWithOutput("npm", ["--prefix", dirPath, "run", "build"]);
```
Comment (joehan) [inline src/extensions/runtimes/node.ts:486]: Nit: Since we don't use the output, should we be using wrapSpawn in this file instead?

**Review by joehan — APPROVED** [review]
LGTM with some nits - thanks for throwing in some extra code cleanup too!

File: src/init/features/genkit/index.ts (hunk 1 above)
Comment (ifielker) [reply src/init/features/genkit/index.ts:71]: Done.

File: src/extensions/runtimes/node.ts (hunk 2 above)
Comment (ifielker) [reply src/extensions/runtimes/node.ts:486]: wrapSpawn requires project directory which is not available here. We may not use the output, but it doesn't really matter. Maybe in future I can refactor wrapSpawn to have project directory be optional.

**Pull request merged**